homeassistant/components/tplink/light.py

```diff
@@ -72,24 +74,31 @@ def __init__(self, smartbulb) -> None:
     @property
     def unique_id(self):
         """Return a unique ID."""
-        return self._sysinfo["mac"]
+        if self.tplink_device.source == SOURCE_CONFIG:
```

Comment:
We shouldn't do this. We should return the same unique ID regardless of the configuration source. 

Using host is also not allowed as IP addresses are not considered unique.

Reply:
I figured that on home assistant's local network, the ip is unique enough, especially if it's static which is what users will have to do in order to make the config work.

I had considered a way to delay adding thy entity until it was available, but abandoned the thought as it didn't look like that was a supported practice.

Another thought was too delay setting the unique id until the entity is available but that sounds awfully close to changing of unique ids. Which is discouraged.

The problem is through unique id is coming from the mac address which isn't available yet because the device is offline.

What ideas can you suggest to solve this problem?

Could adding a unique id to the config be an acceptable solution? That one sounds gross.

Reply:
Unique ID is only checked when the entity is added to Home Assistant.

One solution could be to schedule a timer to check for the offline devices in a bit. Then query the entity registry to see which entities belong to the config entry that we're setting up, and initialize those entities with those unique IDs and mark them as unavailable.